projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/execution/JobExecutionService.java

```diff
@@ -273,7 +273,31 @@ public DataJobExecution readJobExecution(String teamName, String jobName, String
               .lastDeployedDate(jobExecution.getDeployedDate())
               .lastDeployedBy(jobExecution.getDeployedBy())
               .build();
-      return Optional.of(jobExecutionRepository.save(dataJobExecution));
+      jobExecutionRepository.save(dataJobExecution);
+   }
+
+   /**
+    * Returns the last execution of the data job with the specified name, or an empty optional if there
+    * are no executions. The last execution is considered the one with the most recent start time.
+    *
+    * @param dataJobName The name of the data job whose last execution to return.
+    * @return The last execution of the data job if any, otherwise, {@link Optional#empty()}.
+    */
+   public Optional<com.vmware.taurus.service.model.DataJobExecution> getLastExecution(String dataJobName) {
+      return jobExecutionRepository.findFirstByDataJobNameOrderByStartTimeDesc(dataJobName);
+   }
+
+   /**
+    * Returns the last completed execution of the data job with the specified name, or an empty optional if there
+    * are no completed executions. The last completed execution is considered the one with the most recent
+    * end time and status of either FINISHED or FAILED.
+    *
+    * @param dataJobName The name of the data job whose last completed execution to return.
+    * @return The last completed execution of the data job if any, otherwise, {@link Optional#empty()}.
+    */
+   public Optional<com.vmware.taurus.service.model.DataJobExecution> getLastCompletedExecution(String dataJobName) {
```

Comment:
I don't think we need an additional method for loading of execution since in JobService.java:226 we have DataJob.lastExecutionEndTime (old) and DataJobExecution.endTIme(new), so we can update DataJob.lastExecution  based on them.

Reply:
Can you elaborate on this one?